forte/data/extractor/base_extractor.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from abc import ABC
+import logging
+from typing import Tuple, Set, List, Dict, Any
+from typing import Union, Type, Hashable, Iterable, Optional
+from ft.onto.base_ontology import Annotation
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.vocabulary import Vocabulary
+from forte.data.converter.feature import Feature
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "BaseExtractor"
+]
+
+
+class BaseExtractor(ABC):
+    """The functionality of this class is as followed,
+        1. Build vocabulary.
+        2. Extract feature from datapack.
+        3. Add prediction back to datapack.
+
+    Args:
+        config: An instance of `Dict` or
+            :class:`forte.common.configuration.Config` that provides all
+            configurable options. See :meth:`default_configs` for available
+            options and default values.
+            Required keys:
+                entry_type: Type[Entry]. The ontology type that the extractor
+                    get feature from.
+    """
+    def __init__(self, config: Union[Dict, Config]):
+
+        self.config = Config(config, self.default_configs(),
+                                allow_new_hparam=True)
+
+        if not hasattr(self.config, "entry_type"):
```

Comment:
```suggestion
        if "entry_type" not in self.config:
```
Probably using `in` is better.

Reply:
I am not sure wether this is compatible to the Texar Config class. Actually I think if I can just use Dict instead of Config, it will be better. Maybe less dependency is better.

Reply:
If this doesn't work with Texar then it is fine now.